### PR TITLE
Update tdi pipeline builder param name

### PR DESCRIPTION
**Pull request by 5abeel**
Update tdi pipeline builder parameter name from `bf_pipeline_config_binary_file` to `tdi_pipeline_config_binary_file`

### CODE REVIEW

**Review by aashishkuma — APPROVED** [review]
This needs to go along with documentation update in our P4 CP user guide wherever we have references to -tdi_pipeline_builder. LGTM, otherwise.

File: stratum/hal/bin/tdi/tdi_pipeline_builder.cc
Comment (ffoulkes) [inline stratum/hal/bin/tdi/tdi_pipeline_builder.cc]:
```suggestion
// Copyright 2022-2024 Intel Corporation
```

File: stratum/hal/bin/tdi/tdi_pipeline_builder.cc
Comment (ffoulkes) [inline stratum/hal/bin/tdi/tdi_pipeline_builder.cc:1]: This change will affect all TDI targets, not just ES2K. It's a breaking change, since it requires a change on the part of the user. I consider this acceptable, so long as it doesn't break any non-TDI platforms.

**Comment by ffoulkes** [comment]
Ruh-roh. This change appears to have broken the unit test. If you have to make a change to the test (or the target-specific hal file), you'll probably have to do the same thing for all three TDI targets.

**Review by ffoulkes — APPROVED** [review]
Blocking this commit until we're certain that all three TDI targets build and their unit tests pass.

File: stratum/hal/bin/tdi/tdi_pipeline_builder.cc
Comment (5abeel) [reply stratum/hal/bin/tdi/tdi_pipeline_builder.cc]: Looks like other authors didnt update when they touched the file in 2023. Hence, why I used a comma, thinking it was not edited since 2022.
Looking at history, looks like it was edited in 2023 too, so updating to `2022-2024`.

**Comment by 5abeel** [comment]
Made some minor edits and pipeline passes now (wasnt expecting -- so, original failure was perhaps due to randomness in the DpdkHalTest.SanityCheckFailureWhenExtURLsAreInvalid test case?)

**Comment by ffoulkes** [comment]
`BfPipelineConfig` is the name of a Protobuf message. It has not been changed.

**Comment by 5abeel** [comment]
> `BfPipelineConfig` is the name of a Protobuf message. It has not been changed.

Right. This looks like a major feature work -- creating new proto files (since original proto is `hal/lib/barefoot/bf.proto`)

It appears that we are only changing user-exposed names from bf_ to tdi_ or ipu_ as of now and more work is needed to clean the rest of the repository

**Comment by ffoulkes** [comment]
> > `BfPipelineConfig` is the name of a Protobuf message. It has not been changed.
> 
> Right. This looks like a major feature work -- creating new proto files (since original proto is `hal/lib/barefoot/bf.proto`)
> 
> It appears that we are only changing user-exposed names from bf_ to tdi_ or ipu_ as of now and more work is needed to clean the rest of the repository

When the Barefoot platform code was copied and hacked to create the TDI platform code, they renamed `BfDeviceConfig` to `TdiDeviceConfig`, but they didn't rename `BfPipelineConfig`. This may have been an oversight (there were losts of those), or it may have been done intentionally, due, perhaps, to references in other software packages we don't know about.

My advice would be to leave it as-is for now, and to restore the `TdiPipelineConfig` changes you made to their original form.

**Comment by ffoulkes** [comment]
> Made some minor edits and pipeline passes now (wasnt expecting -- so, original failure was perhaps due to randomness in the DpdkHalTest.SanityCheckFailureWhenExtURLsAreInvalid test case?)

It's a really simple test. I don't see how anything shy of a memory corruption issue could have caused it to fail.

**Comment by 5abeel** [comment]
> My advice would be to leave it as-is for now, and to restore the `TdiPipelineConfig` changes you made to their original form.

To clarify, you are suggesting that we do *not* change the parameter from -bf_pipeline_config_binary_file to -tdi_pipeline_config_binary_file?

**Comment by ffoulkes** [comment]
> > My advice would be to leave it as-is for now, and to restore the `TdiPipelineConfig` changes you made to their original form.
> 
> To clarify, you are suggesting that we do _not_ change the parameter from -bf_pipeline_config_binary_file to -tdi_pipeline_config_binary_file?

No, I'm suggesting that you not change `BfPipelineConfig` to `TdiPipelineConfig` in the help text for the flag, because the Protobuf message has not been renamed to `TdiPipelineConfig`. I have no problem with your renaming the command-line flag.

Don't say we renamed the message when we ain't done the work, and don't volunteer to rename it, because we're to smart too pull a stunt like that without seriously considering the repercussions.

**Comment by 5abeel** [comment]
ah ok. got it.
In that case, can I get an approval. I am sitll not sure if its a good idea to merge this late in the cycle -- will wait for a clearance from PM before merging...but lets have the approval completed.

**Comment by ffoulkes** [comment]
> ah ok. got it. In that case, can I get an approval. I am sitll not sure if its a good idea to merge this late in the cycle -- will wait for a clearance from PM before merging...but lets have the approval completed.

Okay. Will do.

**Review by ffoulkes — APPROVED** [review]
LGTM

**Pull request merged**
